lib/routes/user/home/flip_transition.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:flutter/material.dart';
+import 'dart:math';
+import 'dart:async';
+
+class FlipTransition extends StatefulWidget {
+  Widget firstChild;
+  Widget secondChild;
+  Widget thirdChild;
+
+  FlipTransition(this.firstChild, this.secondChild, this.thirdChild);
+
+  @override
+  State<StatefulWidget> createState() {
+    return FlipTransitionState();
+  }
+}
+
+class FlipTransitionState extends State<FlipTransition>
+    with TickerProviderStateMixin {
+  AnimationController _flipAnimationController;
+  Animation _flipAnimation;
+  static const flipDuration = Duration(seconds: 2);
+
+  @override
+  void initState() {
+    super.initState();
+    _flipAnimationController =
+        new AnimationController(vsync: this, duration: flipDuration);
+    _flipAnimation = Tween(begin: 0.0, end: 1.0).animate(CurvedAnimation(
+        parent: _flipAnimationController,
+        curve: Interval(0.0, 1.0, curve: Curves.fastOutSlowIn)));
+
+    _flipAnimationController.addStatusListener((status) {
+      if (status == AnimationStatus.completed) {
+        _flipAnimationController.reverse();
+      }
+    });
+    new Timer(new Duration(milliseconds: 900), () {
```

Comment:
I think the use of timers here is not a good idea. Since there is already a running animation, we can use its value in the build method to render the right child.

Reply:
I also removed the thirdChild parameter.